daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -252,7 +318,7 @@ object SBuiltin {
   }
 
   final case object SBFromTextDecimal extends SBuiltin(1) {
-    private val pattern = """[+-]?[0-9]+(\.[0-9]+)?""".r.pattern
+    private val pattern = """[+-]?\d+(\.\d+)?""".r.pattern
```

Comment:
BTW, are you sure that doesn't allow unicode digits?

Reply:
The answer is no: https://github.com/digital-asset/daml/pull/1429#discussion_r288095576